scripts/utils/printing_tools.js

```diff
@@ -0,0 +1,55 @@
+const BN = require("bn.js")
+const bnOne = new BN(1)
+const bnTwo = new BN(2)
+const bnTen = new BN(10)
+const bn256 = new BN(256)
+const bnMaxUint = bnTwo.pow(bn256).sub(bnOne)
+
+/**
+ * A generalized version of "toWei" for tokens with an arbitrary amount of decimals.
+ * If the decimal representation has more decimals than the maximum amount possible, then the extra decimals are truncated.
+ * @param {string} amount User-friendly representation for the amount of some ERC20 token
+ * @param {integer} decimals Maximum number of decimals of the token
+ * @return {string} decimal representation of the number of token units corresponding to the input amount
+ */
+const fromUserToMachineReadable = function (amount, decimals) {
```

Comment:
This function seems to be quite complicated.

Would the following work as well?

```
return BN((amount.toString()+`e`+ decimals).toLocaleString('fullwide', {useGrouping:false}))
```

Reply:
Mmh, I wasn't able to make it work, but I'd expect it has the same precision problem, since the \`e\` notation hints to some float conversion taking place.
I expect a bad input to be: "222.222222222222222002", 18 decimals.
In fact, running the node console:
```
> parseFloat("222.222222222222222002e18").toString()
  '222222222222222230000'
```

Reply:
okay, I was also not able to find a better solution. Let's go with yours